tests/test_templates.py

```diff
@@ -84,17 +86,4 @@ def test_get_TestEnvironment(self):
                         with open(job.fn('script_{}.sh'.format(op))) as file:
                             reference.extend([msg] + file.read().splitlines())
 
-            self.assertEqual('\n'.join(reference), '\n'.join(generated))
-
-
-# TestCase factory
```

Comment:
Since you removed the test case factory, should this be individual sub tests now?